thirdparty/imgui/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+cmake_minimum_required(VERSION 3.24 FATAL_ERROR)
+# Create the imgui libraries here since the submodule doesn't contain a CMakeLists.txt
+
+set(IMGUI_SOURCES
+  imgui/imgui_draw.cpp
+  imgui/imgui.cpp
+  imgui/imgui_demo.cpp
+  imgui/imgui_draw.cpp
+  imgui/imgui_tables.cpp
+  imgui/imgui_widgets.cpp
+  imgui/backends/imgui_impl_sdl2.cpp
+  imgui_impl_opengl3.cpp
+  )
+
+set(IMGUI_HEADERS
+  imgui/imconfig.h
+  imgui/imgui.h
+  imgui/imgui_internal.h
+  imgui/imstb_rectpack.h
+  imgui/imstb_textedit.h
+  imgui/imstb_truetype.h
+  imgui/backends/imgui_impl_sdl2.h
+  imgui_impl_opengl3.h
+  )
+add_library(imgui STATIC ${IMGUI_SOURCES} ${IMGUI_HEADERS})
+target_link_libraries(imgui PRIVATE SDL2::SDL2OE glad)
+target_include_directories(imgui PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/imgui)
```

Comment:
What's the preferred way to include imgui headers? With <imgui/> prefix, or without? Let's keep a single option, not both.

Reply:
Not sure which one is the preferred way but the backends files ( the one provided with the lib ), like the sdl2 header, expect to include imgui like this:
`#include "imgui.h"`
But I can't keep only `${CMAKE_CURRENT_SOURCE_DIR}/imgui` because the custom backend for openGL I copied is located in `${CMAKE_CURRENT_SOURCE_DIR}`.

Once we port only the files we need and patch them we can probably use one or the other path as the right include directory.

Reply:
Makes sense. Let's leave a todo.